python/hail/methods/statgen.py

```diff
@@ -2908,15 +2909,104 @@ def filter(self):
                 self._left_aligned, self._keep_star))
         return cleanup(m)
 
-
 @typecheck(ds=MatrixTable,
-           n_cores=int,
            r2=numeric,
            window=int,
            memory_per_core=int)
-def ld_prune(ds, n_cores, r2=0.2, window=1000000, memory_per_core=256):
-    jmt = Env.hail().methods.LDPrune.apply(ds._jvds, n_cores, r2, window, memory_per_core)
-    return MatrixTable(jmt)
+def ld_prune(ds, r2=0.2, window=1000000, memory_per_core=256):
+    """Prune variants in linkage disequilibrium.
+
+    Notes
+    -----
+
+    This method prunes variants in linkage disequilibrium in two stages. The first stage
+    is a local pruning step, which prunes variants in the same partition. The parallelism
+    in this step will be affected by the number of partitions in the dataset, as well as
+    the memory per core, which is used to calculate a queue size for the local prune.
+
+    The second stage is a global pruning step which makes use of a correlation matrix
+    across all remaining variants. In the global pruning step, correlated variants are
+    passed to a method that computes a maximal independent set of those variants.
+
+    Parameters
+    ----------
+    ds : :class:`.MatrixTable`
+        dataset to prune
+    r2 : :obj:`float`
+        correlation threshold (exclusive) above which variants are removed
+    window : :obj:`int`
+        distance in kilobases; correlation between variants further apart is not considered in pruning
+    memory_per_core : :obj:`int`
+        memory in MB per core for local pruning
+
+    Returns
+    -------
+    :class:`.Table`
+        Table of variants after pruning.
+    """
+    bytes_per_core = memory_per_core * 1024 * 1024
+    fraction_memory_to_use = 0.25
+    variant_byte_overhead = 50
+    genotypes_per_pack = 32
+    n_samples = ds.count_cols()
+    min_memory_per_core = math.ceil((1 / fraction_memory_to_use) * 8 * n_samples + variant_byte_overhead)
+    if bytes_per_core < min_memory_per_core:
+        raise ValueError("memory per core must be greater than {} MB".format(min_memory_per_core * 1024 * 1024))
+    bytes_per_variant = math.ceil(8 * n_samples / genotypes_per_pack) + variant_byte_overhead
+    memory_available_per_core = bytes_per_core * fraction_memory_to_use
+    max_queue_size = int(max(1.0, math.ceil(memory_available_per_core / bytes_per_variant)))
+
+    sites_only_table = Table(Env.hail().methods.LocalLDPrune.apply(
+        require_biallelic(ds, 'ld_prune')._jvds, float(r2), window, max_queue_size))
+
+    sites_path = new_temp_file()
+    sites_only_table.write(sites_path, overwrite=True)
+    sites_only_table = hl.read_table(sites_path)
+
+    locally_pruned_ds = ds.annotate_rows(pruned=sites_only_table[ds.row_key])
```

Comment:
This currently requires sites_only_tables to get shuffled here.  @patrick-schultz is adding support for ordered tables as we speak.  Alternatively, there is a hacky way to avoid this now (make sites_only_table a sites_only matrix table.)  Let's check in with @patrick-schultz on Monday  and make a strategy for this.